src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -2,72 +2,45 @@
-using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.SyncNamespace
 {
     [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.SyncNamespace), Shared]
     internal sealed class CSharpSyncNamespaceCodeRefactoringProvider 
         : AbstractSyncNamespaceCodeRefactoringProvider<NamespaceDeclarationSyntax, CompilationUnitSyntax, MemberDeclarationSyntax>
     {
-        protected override async Task<SyntaxNode> TryGetApplicableInvocationNode(Document document, int position, CancellationToken cancellationToken)
+        protected override async Task<SyntaxNode> TryGetApplicableInvocationNodeAsync(Document document, TextSpan span, CancellationToken cancellationToken)
         {
-            var compilationUnit = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false) as CompilationUnitSyntax;
-
-            // Here's conditions that trigger the refactoring (all have to be true in each scenario):
-            // 
-            // - There's only one namespace declaration in the document and all types are declared in it:
-            //    1. No nested namespace declaration (even it's empty).
-            //    2. The cursor is on the name of the namespace declaration.
-            //    3. The name of the namespace is valid (i.e. no errors).
-            //    4. No partial type declared in the namespace. Otherwise its multiple declaration will
-            //       end up in different namespace.
-            //
-            // - There's no namespace declaration and all types in the document are declared in global namespace:
-            //    1. The cursor is on the name of first declared type.
-            //    2. No partial type declared in the document. Otherwise its multiple declaration will
-            //       end up in different namespace.
-
-            var triggeringNode = GetTriggeringNode(compilationUnit, position);
-            if (triggeringNode != null)
+            if (!span.IsEmpty)
             {
-                var containsPartial = await ContainsPartialTypeWithMultipleDeclarationsAsync(document, triggeringNode, cancellationToken)
-                    .ConfigureAwait(false);
-
-                if (!containsPartial)
-                {
-                    return triggeringNode;
-                }
+                return null;
             }
 
-            return default;
-        }
+            var position = span.Start;
 
-        private static SyntaxNode GetTriggeringNode(CompilationUnitSyntax compilationUnit, int position)
-        {
+            var compilationUnit = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false) as CompilationUnitSyntax;
             var namespaceDecls = compilationUnit.DescendantNodes(n => n is CompilationUnitSyntax || n is NamespaceDeclarationSyntax)
                 .OfType<NamespaceDeclarationSyntax>().ToImmutableArray();
```

Comment:
repeat of the CompilationUnit || NamespcaeDecl code.  Repeat of hte "find me the namespace decls" code.

Reply:
Yes, but I don't want to promote this to SyntaxFacts until we have a need for this in VB service, and there's no good place to share it between the refactoring and service. I think it's Ok since it's just a one liner.

Reply:
> and there's no good place to share it between the refactoring and service.

Can tehre not just be a sibling class with helpers that are shared?  That's what we generally do elsewhere...

Reply:
Sure. I just thought it's a overkill for this.

Reply:
That's fair.  I don't care that much either :)  Just something i noticed :)